### PR TITLE
Add Fvar Function with Test to LightCurve

### PR DESCRIPTION
**Pull request by cnachi**
Please dont merge yet ; will provide a nicer example test

### CODE REVIEW

File: gammapy/time/lightcurve.py [hunk 1]
```diff
@@ -68,3 +69,35 @@ def simulate_example(cls):
         lc['FLUX_ERR'] = Quantity([0.1, 0.4, 0.7, 0.9], 'cm^-2 s^-1')
 
         return lc
+
+    def compute_fvar(self):
+        """Calculate the fractional excess variance (Fvar) of `LightCurve`.
+
```
Comment (cdeil) [inline gammapy/time/lightcurve.py]: Please add a link to a paper or webpage with a description / formula for this.

Maybe also mention which columns from the lightcurve are accessed.

Also put this in the docstring:
```
Returns
--------
fvar, fvarerr : float
    Normalised excess variance and error on it.
```
```

File: gammapy/time/lightcurve.py [hunk 2]
```diff
@@ -68,3 +69,35 @@ def simulate_example(cls):
...
+        """
+        time_min = self['TIME_MIN']
+        time_max = self['TIME_MAX']
```
Comment (cdeil) [inline gammapy/time/lightcurve.py]: Looks like you have mixed tabs and spaces?
Should all be spaces and 4 space indent.

pep8 tool will tell you about it, autopep8 tool can auto-fix it.
(or configure your editor for Python dev)

File: gammapy/time/lightcurve.py [hunk 3]
```diff
@@ -68,3 +69,35 @@ def simulate_example(cls):
...
+   	Ssq = (np.sum( (flux[:]-phimean*unityrow[:])**2) ) / (nptsperbin-1.0)
+   	S = np.sqrt(Ssq)
+   	Sigsq = np.nansum(fluxerr[:]**2) / nptsperbin
+   	#print fluxerr,np.nansum(fluxerr)#,nptsperbin        
+   	fvar = np.sqrt(np.abs(Ssq-Sigsq)) / phimean
+   	fluxerrT = np.transpose(fluxerr)
+   	#ascii.write([fluxerrT],'simfluxerr.txt')#,np.sum(fluxerr),fluxerr#nptsperbin
+   	sigxserrA = np.sqrt( 2.0/nptsperbin) * (Sigsq/phimean)**2
+   	sigxserrB = np.sqrt(Sigsq/nptsperbin) * (2.0*fvar/phimean)
+	#print sigxserrA,sigxserrB
```
Comment (cdeil) [inline gammapy/time/lightcurve.py]: Remove print & IPython after debugging.

File: gammapy/time/tests/test_lightcurve.py [hunk 4]
```diff
@@ -3,7 +3,9 @@
 from astropy.units import Quantity
 from astropy.tests.helper import assert_quantity_allclose
 from ...utils.testing import requires_dependency
+#import matplotlib.pyplot as plt
```
Comment (cdeil) [inline gammapy/time/tests/test_lightcurve.py]: Remove

File: gammapy/time/lightcurve.py (hunk 1 above)
Comment (cdeil) [inline gammapy/time/lightcurve.py]: If you have time, you could give a formula or example in the method docstring like here:

http://docs.gammapy.org/en/latest/api/gammapy.stats.significance.html
http://docs.gammapy.org/en/latest/_modules/gammapy/stats/poisson.html#significance

**Comment by cdeil** [comment]
@cnachi - Do you have time to finish this up?

It's always best to try and get pull requests in quickly, otherwise it's usually hard to re-start and there's merge conflicts if others touched the file etc.
If you want, you can also come by my office and we finish it together.

File: gammapy/time/lightcurve.py (hunk 3 above)
Comment (cnachi) [inline gammapy/time/lightcurve.py]: Removed comments

File: gammapy/time/lightcurve.py (hunk 2 above)
Comment (cnachi) [inline gammapy/time/lightcurve.py]: Is this the right place to get autopep8 ?
https://github.com/tell-k/vim-autopep8

**Comment by cdeil** [comment]
@cnachi - Do you have time to finish this up? Or should I just do it?

I think we discussed it and it was mostly a matter of improving the docstring and a little bit the method (e.g. `unityrow = np.ones(nptsperbin)` isn't needed because Numpy broadcasting will take care of that automatically).

**Comment by cnachi** [comment]
@cdeil : I committed some changes a few days ago  - I forgot to alert you - I basically edited the docstring to add the formula and reference and also used the numpy broadcasting. Can you see this ?

**Comment by cdeil** [comment]
@cnachi - No, I don't see a formula in the docstring.

Each pull request has three "views" you / I / anyone can choose at the top:

- "Conversation" at `https://github.com/gammapy/gammapy/pull/798`
- "Commits" at `https://github.com/gammapy/gammapy/pull/798/commits`
- "Files" at `https://github.com/gammapy/gammapy/pull/798/files` showing the diff

You could check the local commits using `git log` versus the commits on this pull request branch on github. Maybe you have local commits that you didn't push to github yet?

And yes, please after adressing some comments and pushing new commits on a PR, please leave another comment an status, i.e. whether you think the PR is ready or if you have some question, so that I know what to do.

**Comment by cnachi** [comment]
@cdeil : Ahh I think you should be able to see them now

**Comment by cdeil** [comment]
@cnachi - Thank you!

I've cleaned up the docstring and implementation a bit in 9d9be3b3 and then merged this.
Key changes are:
- start docstring with `r"""`, i.e. have a "raw string" to prevent Python from processing escape characters like `\`. That's needed when you have LaTeX in a string.
- don't access `TIME` columns in the method. They aren't used to compute f_var.

**Comment by cnachi** [comment]
@cdeil : Thanks for cleaning up - where does this appear in the docs of gammapy ?

**Comment by cdeil** [comment]
It should appear here: http://docs.gammapy.org/en/latest/api/gammapy.time.LightCurve.html

It doesn't at the moment because of a docs build issue: http://readthedocs.org/projects/gammapy/builds/4812909/

I'll investigate and fix in the coming days.

**Comment by cdeil** [comment]
@cnachi - just FYI: the online docs build issue has been resolved in the meantime.

Docs for the new method now are here:
http://docs.gammapy.org/en/latest/api/gammapy.time.LightCurve.html#gammapy.time.LightCurve.compute_fvar

**Comment by cnachi** [comment]
@cdeil : Great thanks !

**Pull request merged**